From 3e898010702477dfa5c32896440d445beb544aaf Mon Sep 17 00:00:00 2001
From: classabbyamp <dev@placeviolette.net>
Date: Mon, 17 Aug 2026 18:48:33 -0400
Subject: [PATCH] build udev as a standalone package

* prevents unnecessary components from building
* renames systemd-udevd to udevd
* renames systemd-hwdb to udev-hwdb
* removes references to documentation that doesn't get built
---
 factory/templates/meson.build                 |   4 +-
 hwdb.d/60-autosuspend.hwdb                    |   2 +-
 hwdb.d/60-evdev.hwdb                          |   2 +-
 hwdb.d/60-input-id.hwdb                       |   2 +-
 hwdb.d/60-keyboard.hwdb                       |   2 +-
 hwdb.d/60-sensor.hwdb                         |   2 +-
 hwdb.d/70-av-production.hwdb                  |   2 +-
 hwdb.d/70-hardware-wallets.hwdb               |   2 +-
 hwdb.d/70-joystick.hwdb                       |   2 +-
 hwdb.d/70-lights.hwdb                         |   2 +-
 hwdb.d/70-mouse.hwdb                          |   2 +-
 hwdb.d/70-pointingstick.hwdb                  |   2 +-
 hwdb.d/70-software-radio.hwdb                 |   2 +-
 hwdb.d/70-sound-card.hwdb                     |   2 +-
 hwdb.d/70-touchpad.hwdb                       |   2 +-
 hwdb.d/80-ieee1394-unit-function.hwdb         |   2 +-
 hwdb.d/README                                 |   7 +-
 hwdb.d/meson.build                            |   2 +-
 man/hwdb.xml                                  |   4 +-
 man/meson.build                               |  37 +--
 man/rules/meson.build                         |   2 +-
 man/{systemd-hwdb.xml => udev-hwdb.xml}       |  19 +-
 man/udev.conf.xml                             |  12 +-
 man/udev.xml                                  |  40 +--
 man/udev_device_get_syspath.xml               |   1 -
 man/udev_device_has_tag.xml                   |   1 -
 man/udev_device_new_from_syspath.xml          |   1 -
 man/udev_enumerate_add_match_subsystem.xml    |   1 -
 man/udev_enumerate_new.xml                    |   1 -
 man/udev_enumerate_scan_devices.xml           |   1 -
 man/udev_list_entry.xml                       |   1 -
 man/udev_monitor_filter_update.xml            |   1 -
 man/udev_monitor_new_from_netlink.xml         |   1 -
 man/udev_monitor_receive_device.xml           |   1 -
 man/udev_new.xml                              |  10 -
 man/udevadm.xml                               |  82 ++----
 meson.build                                   | 265 +-----------------
 rules.d/README                                |   7 +-
 rules.d/meson.build                           |   8 -
 shell-completion/bash/meson.build             |  43 +--
 .../bash/{systemd-hwdb => udev-hwdb}          |   6 +-
 .../zsh/{_systemd-hwdb => _udev-hwdb}         |   4 +-
 shell-completion/zsh/_udevadm                 |  14 +-
 shell-completion/zsh/meson.build              |  36 +--
 src/core/meson.build                          |  15 -
 src/hwdb/hwdb.c                               |   2 +-
 src/hwdb/meson.build                          |   2 +-
 src/libsystemd/sd-hwdb/sd-hwdb.c              |   2 +-
 src/test/meson.build                          |  25 +-
 src/udev/meson.build                          |   2 +-
 src/udev/udev-config.c                        |   2 -
 src/udev/udevadm-hwdb.c                       |   4 +-
 src/udev/udevadm-wait.c                       |   2 +-
 src/udev/udevd.c                              |   2 +-
 test/meson.build                              |   2 +-
 55 files changed, 121 insertions(+), 581 deletions(-)
 rename man/{systemd-hwdb.xml => udev-hwdb.xml} (77%)
 rename shell-completion/bash/{systemd-hwdb => udev-hwdb} (94%)
 rename shell-completion/zsh/{_systemd-hwdb => _udev-hwdb} (91%)

diff --git a/factory/templates/meson.build b/factory/templates/meson.build
index 0f53217a48..02f5ba5d03 100644
--- a/factory/templates/meson.build
+++ b/factory/templates/meson.build
@@ -6,14 +6,14 @@ custom_target(
         input : 'locale.conf.in',
         output : 'locale.conf',
         command : [jinja2_cmdline, '@INPUT@', '@OUTPUT@'],
-        install : true,
+        install : false,
         install_dir : factory_etc_dir)
 
 custom_target(
         input : 'vconsole.conf.in',
         output : 'vconsole.conf',
         command : [jinja2_cmdline, '@INPUT@', '@OUTPUT@'],
-        install : true,
+        install : false,
         install_dir : factory_etc_dir)
 
 if conf.get('ENABLE_NSS') == 1
diff --git a/hwdb.d/60-autosuspend.hwdb b/hwdb.d/60-autosuspend.hwdb
index 0e38c0a9a4..73127ff0f9 100644
--- a/hwdb.d/60-autosuspend.hwdb
+++ b/hwdb.d/60-autosuspend.hwdb
@@ -11,7 +11,7 @@
 # To add local entries, create a new file
 #   /etc/udev/hwdb.d/61-autosuspend-local.hwdb
 # and add your rules there. To load the new rules execute (as root):
-#   systemd-hwdb update
+#   udev-hwdb update
 #   udevadm trigger /dev/…
 #
 # If your changes are generally applicable, send them as a pull request to
diff --git a/hwdb.d/60-evdev.hwdb b/hwdb.d/60-evdev.hwdb
index e1e30700c4..a2ef574fdd 100644
--- a/hwdb.d/60-evdev.hwdb
+++ b/hwdb.d/60-evdev.hwdb
@@ -33,7 +33,7 @@
 # To add local entries, create a new file
 #   /etc/udev/hwdb.d/61-evdev-local.hwdb
 # and add your rules there. To load the new rules execute (as root):
-#   systemd-hwdb update
+#   udev-hwdb update
 #   udevadm trigger /dev/input/eventXX
 # where /dev/input/eventXX is the device in question. If in doubt, simply use
 # /dev/input/event* to reload all input rules.
diff --git a/hwdb.d/60-input-id.hwdb b/hwdb.d/60-input-id.hwdb
index 700e7ee264..ae19094e50 100644
--- a/hwdb.d/60-input-id.hwdb
+++ b/hwdb.d/60-input-id.hwdb
@@ -10,7 +10,7 @@
 # To add local entries, create a new file
 #   /etc/udev/hwdb.d/61-input-id-local.hwdb
 # and add your rules there. To load the new rules execute (as root):
-#   systemd-hwdb update
+#   udev-hwdb update
 #   udevadm trigger /dev/input/eventXX
 # where /dev/input/eventXX is the device in question. If in
 # doubt, simply use /dev/input/event* to reload all input rules.
diff --git a/hwdb.d/60-keyboard.hwdb b/hwdb.d/60-keyboard.hwdb
index 08be116265..0521871923 100644
--- a/hwdb.d/60-keyboard.hwdb
+++ b/hwdb.d/60-keyboard.hwdb
@@ -46,7 +46,7 @@
 # To add local entries, create a new file
 #   /etc/udev/hwdb.d/61-keyboard-local.hwdb
 # and add your rules there. To load the new rules execute (as root):
-#   systemd-hwdb update
+#   udev-hwdb update
 #   udevadm trigger /dev/input/eventXX
 # where /dev/input/eventXX is the keyboard in question. If in doubt, simply use
 # /dev/input/event* to reload all input rules.
diff --git a/hwdb.d/60-sensor.hwdb b/hwdb.d/60-sensor.hwdb
index 7b57c18748..7d00e43e96 100644
--- a/hwdb.d/60-sensor.hwdb
+++ b/hwdb.d/60-sensor.hwdb
@@ -23,7 +23,7 @@
 # To add local entries, create a new file
 #   /etc/udev/hwdb.d/61-sensor-local.hwdb
 # and add your rules there. To load the new rules execute (as root):
-#   systemd-hwdb update
+#   udev-hwdb update
 #   udevadm trigger -v -p DEVNAME=/dev/iio:deviceXXX
 # where /dev/iio:deviceXXX is the device in question.
 #
diff --git a/hwdb.d/70-av-production.hwdb b/hwdb.d/70-av-production.hwdb
index 6c0fd6104e..e9f6880d55 100644
--- a/hwdb.d/70-av-production.hwdb
+++ b/hwdb.d/70-av-production.hwdb
@@ -10,7 +10,7 @@
 # To add local entries, create a new file
 #   /etc/udev/hwdb.d/71-av-production-local.hwdb
 # and add your rules there. To load the new rules execute (as root):
-#   systemd-hwdb update
+#   udev-hwdb update
 #   udevadm trigger /dev/…
 #
 # If your changes are generally applicable, send them as a pull request to
diff --git a/hwdb.d/70-hardware-wallets.hwdb b/hwdb.d/70-hardware-wallets.hwdb
index 56d7cc6e23..fb85fc7329 100644
--- a/hwdb.d/70-hardware-wallets.hwdb
+++ b/hwdb.d/70-hardware-wallets.hwdb
@@ -7,7 +7,7 @@
 # To add local entries, create a new file
 #   /etc/udev/hwdb.d/71-hardware-wallets-local.hwdb
 # and add your rules there. To load the new rules execute (as root):
-#   systemd-hwdb update
+#   udev-hwdb update
 #   udevadm trigger /dev/…
 #
 # If your changes are generally applicable, send them as a pull request to
diff --git a/hwdb.d/70-joystick.hwdb b/hwdb.d/70-joystick.hwdb
index ec5006332e..91ba5747a8 100644
--- a/hwdb.d/70-joystick.hwdb
+++ b/hwdb.d/70-joystick.hwdb
@@ -13,7 +13,7 @@
 # To add local entries, create a new file
 #   /etc/udev/hwdb.d/71-joystick-local.hwdb
 # and add your rules there. To load the new rules execute (as root):
-#   systemd-hwdb update
+#   udev-hwdb update
 #   udevadm trigger /dev/input/eventXX
 # where /dev/input/eventXX is the joystick in question. If in
 # doubt, simply use /dev/input/event* to reload all input rules.
diff --git a/hwdb.d/70-lights.hwdb b/hwdb.d/70-lights.hwdb
index af56968abd..7f8d95b7d5 100644
--- a/hwdb.d/70-lights.hwdb
+++ b/hwdb.d/70-lights.hwdb
@@ -9,7 +9,7 @@
 # To add local entries, create a new file
 #   /etc/udev/hwdb.d/71-lights-local.hwdb
 # and add your rules there. To load the new rules execute (as root):
-#   systemd-hwdb update
+#   udev-hwdb update
 #   udevadm trigger /dev/…
 #
 # If your changes are generally applicable, send them as a pull request to
diff --git a/hwdb.d/70-mouse.hwdb b/hwdb.d/70-mouse.hwdb
index 3009ea2fdc..51bcac6b04 100644
--- a/hwdb.d/70-mouse.hwdb
+++ b/hwdb.d/70-mouse.hwdb
@@ -28,7 +28,7 @@
 # To add local entries, create a new file
 #   /etc/udev/hwdb.d/71-mouse-local.hwdb
 # and add your rules there. To load the new rules execute (as root):
-#   systemd-hwdb update
+#   udev-hwdb update
 #   udevadm trigger /dev/input/eventXX
 # where /dev/input/eventXX is the mouse in question. If in doubt, simply use
 # /dev/input/event* to reload all input rules.
diff --git a/hwdb.d/70-pointingstick.hwdb b/hwdb.d/70-pointingstick.hwdb
index 6ff228bee6..1a160bf508 100644
--- a/hwdb.d/70-pointingstick.hwdb
+++ b/hwdb.d/70-pointingstick.hwdb
@@ -37,7 +37,7 @@
 # To add local entries, create a new file
 #   /etc/udev/hwdb.d/71-pointingstick-local.hwdb
 # and add your rules there. To load the new rules execute (as root):
-#   systemd-hwdb update
+#   udev-hwdb update
 #   udevadm trigger /dev/input/eventXX
 # where /dev/input/eventXX is the pointingstick in question. If in doubt, simply
 # use /dev/input/event* to reload all input rules.
diff --git a/hwdb.d/70-software-radio.hwdb b/hwdb.d/70-software-radio.hwdb
index 63af9c77cd..ddffda8249 100644
--- a/hwdb.d/70-software-radio.hwdb
+++ b/hwdb.d/70-software-radio.hwdb
@@ -8,7 +8,7 @@
 # To add local entries, create a new file
 #   /etc/udev/hwdb.d/71-software-radio-local.hwdb
 # and add your rules there. To load the new rules execute (as root):
-#   systemd-hwdb update
+#   udev-hwdb update
 #   udevadm trigger /dev/…
 #
 # If your changes are generally applicable, send them as a pull request to
diff --git a/hwdb.d/70-sound-card.hwdb b/hwdb.d/70-sound-card.hwdb
index 62d005d61e..dd2ac6e038 100644
--- a/hwdb.d/70-sound-card.hwdb
+++ b/hwdb.d/70-sound-card.hwdb
@@ -7,7 +7,7 @@
 # To add local entries, create a new file
 #   /etc/udev/hwdb.d/71-sound-card-local.hwdb
 # and add your rules there. To load the new rules execute (as root):
-#   systemd-hwdb update
+#   udev-hwdb update
 #   udevadm trigger /dev/…
 #
 # If your changes are generally applicable, send them as a pull request to
diff --git a/hwdb.d/70-touchpad.hwdb b/hwdb.d/70-touchpad.hwdb
index 8c166d7013..152bec79b8 100644
--- a/hwdb.d/70-touchpad.hwdb
+++ b/hwdb.d/70-touchpad.hwdb
@@ -13,7 +13,7 @@
 # To add local entries, create a new file
 #   /etc/udev/hwdb.d/71-touchpad-local.hwdb
 # and add your rules there. To load the new rules execute (as root):
-#   systemd-hwdb update
+#   udev-hwdb update
 #   udevadm trigger /dev/input/eventXX
 # where /dev/input/eventXX is the touchpad in question. If in
 # doubt, simply use /dev/input/event* to reload all input rules.
diff --git a/hwdb.d/80-ieee1394-unit-function.hwdb b/hwdb.d/80-ieee1394-unit-function.hwdb
index 62a8dda90b..38579ce764 100644
--- a/hwdb.d/80-ieee1394-unit-function.hwdb
+++ b/hwdb.d/80-ieee1394-unit-function.hwdb
@@ -21,7 +21,7 @@
 #  - ven0x000000units*0x000000:0x000000*
 #
 # The hexadecimal digits part of the customized key should be lower-case. Linux FireWire subsystem
-# uses lower-case value for attributes of sysfs node, and systemd-hwdb parses the custom key by
+# uses lower-case value for attributes of sysfs node, and udev-hwdb parses the custom key by
 # case-sensitive way. On the other hand, it parses kernel alias by case-insensitive way.
 #
 # The entry should have some of IEEE1394_UNIT_FUNCTION_XXX environment variables to express function
diff --git a/hwdb.d/README b/hwdb.d/README
index 594f5bfe0d..58e9052275 100644
--- a/hwdb.d/README
+++ b/hwdb.d/README
@@ -3,9 +3,6 @@ of mappings from modalias-like keys (which identify specific hardware devices)
 to udev properties.
 
 Files in this directory are not read by udev directly. Instead,
-man:systemd-hwdb(8) compiles them into a binary database.
+man:udev-hwdb(8) compiles them into a binary database.
 
-See man:hwdb(7) for an overview of the configuration file format, and
-man:systemd-udevd.service(8) for a description of the udev daemon.
-
-Use 'systemd-analyze cat-config udev/hwdb.d' to display the effective config.
+See man:hwdb(7) for an overview of the configuration file format.
diff --git a/hwdb.d/meson.build b/hwdb.d/meson.build
index b031bd1d92..61d46f0b2b 100644
--- a/hwdb.d/meson.build
+++ b/hwdb.d/meson.build
@@ -64,7 +64,7 @@ if conf.get('ENABLE_HWDB') == 1
                 install_emptydir(sysconfdir / 'udev/hwdb.d', install_tag : 'hwdb')
 
                 meson.add_install_script(sh, '-c',
-                                         'test -n "$DESTDIR" || @0@/systemd-hwdb update'.format(bindir),
+                                         'test -n "$DESTDIR" || @0@/udev-hwdb update'.format(bindir),
                                          install_tag: 'hwdb')
         endif
 
diff --git a/man/hwdb.xml b/man/hwdb.xml
index fc8aa92ea6..dca25677c7 100644
--- a/man/hwdb.xml
+++ b/man/hwdb.xml
@@ -73,7 +73,7 @@
       higher priority.</para>
 
       <para>The content of all hwdb files is read by
-      <citerefentry><refentrytitle>systemd-hwdb</refentrytitle><manvolnum>8</manvolnum></citerefentry>
+      <citerefentry><refentrytitle>udev-hwdb</refentrytitle><manvolnum>8</manvolnum></citerefentry>
       and compiled to a binary database located at <filename>/etc/udev/hwdb.bin</filename>,
       or alternatively <filename>/usr/lib/udev/hwdb.bin</filename> if you want ship the
       compiled database in an immutable image.  During runtime, only the binary database
@@ -146,7 +146,7 @@ PROPERTY_WITH_SPACES=some string</programlisting>
   <refsect1>
     <title>See Also</title>
     <para><simplelist type="inline">
-      <member><citerefentry><refentrytitle>systemd-hwdb</refentrytitle><manvolnum>8</manvolnum></citerefentry></member>
+      <member><citerefentry><refentrytitle>udev-hwdb</refentrytitle><manvolnum>8</manvolnum></citerefentry></member>
     </simplelist></para>
   </refsect1>
 </refentry>
diff --git a/man/meson.build b/man/meson.build
index 8a5b446f24..d34d42fe39 100644
--- a/man/meson.build
+++ b/man/meson.build
@@ -67,6 +67,16 @@ foreach tuple : manpages
                 endif
         endforeach
 
+        if stem.contains('systemd')
+                have = false
+        endif
+        if not stem.contains('udev')
+                have = false
+        endif
+        if stem == 'hwdb'
+                have = true
+        endif
+
         if have
                 file = files(tuple[0] + '.xml')
                 source_xml_files += file
@@ -126,13 +136,7 @@ systemd_directives_xml = custom_target(
         command : [make_directive_index_py, '@OUTPUT@', '@INPUT@'])
 
 nonindex_xml_files = source_xml_files + [systemd_directives_xml]
-systemd_index_xml = custom_target(
-        input : nonindex_xml_files,
-        output : 'systemd.index.xml',
-        command : [make_man_index_py, '@OUTPUT@'] + nonindex_xml_files)
-
-foreach tuple : xsltproc.found() and have_lxml ? [['systemd.directives', '7', systemd_directives_xml],
-                                                  ['systemd.index',      '7', systemd_index_xml]] : []
+foreach tuple : []
         stem = tuple[0]
         section = tuple[1]
         xml = tuple[2]
@@ -216,25 +220,6 @@ configure_file(
 
 ############################################################
 
-update_dbus_docs = custom_target(
-        output : 'update-dbus-docs',
-        command : [update_dbus_docs_py, '--build-dir', meson.project_build_root(), '@INPUT@'],
-        input : dbus_docs,
-        depends : dbus_programs)
-
-if conf.get('BUILD_MODE_DEVELOPER') == 1
-        test('dbus-docs-fresh',
-             update_dbus_docs_py,
-             suite : 'dist',
-             args : ['--build-dir', meson.project_build_root(), '--test', dbus_docs],
-             depends : dbus_programs)
-
-        test('check-version-history',
-             check_version_history_py,
-             suite : 'dist',
-             args : source_xml_files)
-endif
-
 update_man_rules = custom_target(
         output : 'update-man-rules',
         command : [update_man_rules_py,
diff --git a/man/rules/meson.build b/man/rules/meson.build
index a3d0650e27..20a581e515 100644
--- a/man/rules/meson.build
+++ b/man/rules/meson.build
@@ -1061,7 +1061,7 @@ manpages = [
   'ENABLE_HIBERNATE'],
  ['systemd-homed.service', '8', ['systemd-homed'], 'ENABLE_HOMED'],
  ['systemd-hostnamed.service', '8', ['systemd-hostnamed'], 'ENABLE_HOSTNAMED'],
- ['systemd-hwdb', '8', [], 'ENABLE_HWDB'],
+ ['udev-hwdb', '8', [], 'ENABLE_HWDB'],
  ['systemd-id128', '1', [], ''],
  ['systemd-imds-generator', '8', [], 'ENABLE_IMDS'],
  ['systemd-imds', '1', ['systemd-imds-import.service'], 'ENABLE_IMDS'],
diff --git a/man/systemd-hwdb.xml b/man/udev-hwdb.xml
similarity index 77%
rename from man/systemd-hwdb.xml
rename to man/udev-hwdb.xml
index ab27643e6e..b9a35fb77c 100644
--- a/man/systemd-hwdb.xml
+++ b/man/udev-hwdb.xml
@@ -3,34 +3,34 @@
   "http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd">
 <!-- SPDX-License-Identifier: LGPL-2.1-or-later -->
 
-<refentry id="systemd-hwdb" conditional="ENABLE_HWDB"
+<refentry id="udev-hwdb" conditional="ENABLE_HWDB"
           xmlns:xi="http://www.w3.org/2001/XInclude">
 
   <refentryinfo>
-    <title>systemd-hwdb</title>
+    <title>udev-hwdb</title>
     <productname>systemd</productname>
   </refentryinfo>
 
   <refmeta>
-    <refentrytitle>systemd-hwdb</refentrytitle>
+    <refentrytitle>udev-hwdb</refentrytitle>
     <manvolnum>8</manvolnum>
   </refmeta>
 
   <refnamediv>
-    <refname>systemd-hwdb</refname><refpurpose>hardware database management tool</refpurpose>
+    <refname>udev-hwdb</refname><refpurpose>hardware database management tool</refpurpose>
   </refnamediv>
 
   <refsynopsisdiv>
     <cmdsynopsis>
-      <command>systemd-hwdb <optional>options</optional> update</command>
+      <command>udev-hwdb <optional>options</optional> update</command>
     </cmdsynopsis>
     <cmdsynopsis>
-      <command>systemd-hwdb <optional>options</optional> query <replaceable>modalias</replaceable></command>
+      <command>udev-hwdb <optional>options</optional> query <replaceable>modalias</replaceable></command>
     </cmdsynopsis>
   </refsynopsisdiv>
 
   <refsect1><title>Description</title>
-    <para><command>systemd-hwdb</command> expects a command and command
+    <para><command>udev-hwdb</command> expects a command and command
     specific arguments.  It manages the binary hardware database.</para>
   </refsect1>
 
@@ -66,13 +66,13 @@
       <xi:include href="standard-options.xml" xpointer="help" />
     </variablelist>
 
-    <refsect2><title>systemd-hwdb
+    <refsect2><title>udev-hwdb
       <optional><replaceable>options</replaceable></optional>
       update</title>
       <para>Update the binary database.</para>
     </refsect2>
 
-    <refsect2><title>systemd-hwdb
+    <refsect2><title>udev-hwdb
       <optional><replaceable>options</replaceable></optional>
       query
       <replaceable>MODALIAS</replaceable>
@@ -84,7 +84,6 @@
   <refsect1>
     <title>See Also</title>
     <para>
-      <citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry>
       <citerefentry><refentrytitle>hwdb</refentrytitle><manvolnum>7</manvolnum></citerefentry>
     </para>
   </refsect1>
diff --git a/man/udev.conf.xml b/man/udev.conf.xml
index 5e42bf304c..05c672c556 100644
--- a/man/udev.conf.xml
+++ b/man/udev.conf.xml
@@ -41,7 +41,7 @@
     <title>Description</title>
 
     <para>These files contain configuration options for
-      <citerefentry><refentrytitle>systemd-udevd</refentrytitle><manvolnum>8</manvolnum></citerefentry>.
+      <citerefentry><refentrytitle>udevd</refentrytitle><manvolnum>8</manvolnum></citerefentry>.
       The syntax of these files is very simple: a list
       of assignments, one per line.
       All empty lines or lines beginning with <literal>#</literal> are
@@ -117,7 +117,7 @@
         <term><varname>resolve_names=</varname></term>
 
         <listitem>
-          <para>Specifies when systemd-udevd should resolve names of users and groups.  When set to
+          <para>Specifies when udevd should resolve names of users and groups.  When set to
           <option>early</option> (the default), names will be resolved when the rules are parsed.
           When set to <option>late</option>, names will be resolved for every event.  When set to
           <option>never</option>, names will never be resolved and all devices will be owned by
@@ -133,7 +133,7 @@
         <term><varname>timeout_signal=</varname></term>
 
         <listitem>
-          <para>Specifies a signal that <filename>systemd-udevd</filename> will send on worker
+          <para>Specifies a signal that <filename>udevd</filename> will send on worker
           timeouts. Note that both workers and spawned processes will be killed using this
           signal. Defaults to <option>SIGKILL</option>.</para>
 
@@ -143,16 +143,16 @@
     </variablelist>
 
     <para>
-      In addition, <filename>systemd-udevd</filename> can be configured
+      In addition, <filename>udevd</filename> can be configured
       by command line options and the kernel command line (see
-      <citerefentry><refentrytitle>systemd-udevd</refentrytitle><manvolnum>8</manvolnum></citerefentry>).
+      <citerefentry><refentrytitle>udevd</refentrytitle><manvolnum>8</manvolnum></citerefentry>).
     </para>
   </refsect1>
 
   <refsect1>
     <title>See Also</title>
     <para><simplelist type="inline">
-      <member><citerefentry><refentrytitle>systemd-udevd</refentrytitle><manvolnum>8</manvolnum></citerefentry></member>
+      <member><citerefentry><refentrytitle>udevd</refentrytitle><manvolnum>8</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev</refentrytitle><manvolnum>7</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udevadm</refentrytitle><manvolnum>8</manvolnum></citerefentry></member>
     </simplelist></para>
diff --git a/man/udev.xml b/man/udev.xml
index 9a76db1ecc..b37d0dea05 100644
--- a/man/udev.xml
+++ b/man/udev.xml
@@ -32,8 +32,7 @@
     names provide a way to reliably identify devices based on their properties or
     current configuration.</para>
 
-    <para>The udev daemon, <citerefentry><refentrytitle>systemd-udevd.service</refentrytitle>
-    <manvolnum>8</manvolnum></citerefentry>, receives device uevents directly from
+    <para>The udev daemon, udevd, receives device uevents directly from
     the kernel whenever a device is added or removed from the system, or it changes its
     state. When udev receives a device event, it matches its configured set of rules
     against various device attributes to identify the device. Rules that match may
@@ -279,9 +278,7 @@
                 <varlistentry>
                   <term><literal>arch</literal></term>
                   <listitem>
-                    <para>System's architecture. See <option>ConditionArchitecture=</option> in
-                    <citerefentry><refentrytitle>systemd.unit</refentrytitle><manvolnum>5</manvolnum></citerefentry>
-                    for possible values.</para>
+                    <para>System's architecture.</para>
 
                   <xi:include href="version-info.xml" xpointer="v244"/>
                   </listitem>
@@ -289,9 +286,7 @@
                 <varlistentry>
                   <term><literal>virt</literal></term>
                   <listitem>
-                    <para>System's virtualization environment. See
-                    <citerefentry><refentrytitle>systemd-detect-virt</refentrytitle><manvolnum>1</manvolnum></citerefentry>
-                    for possible values.</para>
+                    <para>System's virtualization environment.</para>
 
                     <xi:include href="version-info.xml" xpointer="v244"/>
                   </listitem>
@@ -299,9 +294,7 @@
                 <varlistentry>
                   <term><literal>cvm</literal></term>
                   <listitem>
-                    <para>System's confidential virtualization technology. See
-                    <citerefentry><refentrytitle>systemd-detect-virt</refentrytitle><manvolnum>1</manvolnum></citerefentry>
-                    for possible values.</para>
+                    <para>System's confidential virtualization technology.</para>
 
                     <xi:include href="version-info.xml" xpointer="v254"/>
                   </listitem>
@@ -410,9 +403,7 @@
           <varlistentry>
             <term><varname>NAME</varname></term>
             <listitem>
-              <para>The name to use for a network interface. See
-              <citerefentry><refentrytitle>systemd.link</refentrytitle><manvolnum>5</manvolnum></citerefentry>
-              for a higher-level mechanism for setting the interface name.
+              <para>The name to use for a network interface.
               The name of a device node cannot be changed by udev, only additional
               symlinks can be created.</para>
             </listitem>
@@ -538,15 +529,10 @@
               a long period of time may block all further events for this or a dependent device.</para>
 
               <para>Note that running programs that access the network or mount/unmount filesystems is not
-              allowed inside of udev rules, due to the default sandbox that is enforced on
-              <filename>systemd-udevd.service</filename>.</para>
+              allowed inside of udev rules, due to the default sandbox that is enforced.</para>
 
               <para>Starting daemons or other long-running processes is not allowed; the forked processes,
-              detached or not, will be unconditionally killed after the event handling has finished. In order
-              to activate long-running processes from udev rules, provide a service unit and pull it in from a
-              udev device using the <varname>SYSTEMD_WANTS</varname> device property. See
-              <citerefentry><refentrytitle>systemd.device</refentrytitle><manvolnum>5</manvolnum></citerefentry>
-              for details.</para>
+              detached or not, will be unconditionally killed after the event handling has finished.</para>
             </listitem>
           </varlistentry>
 
@@ -668,8 +654,8 @@
                     in the directory
                     <filename>/run/udev/static_node-tags/<replaceable>tag</replaceable></filename>
                     pointing at the static device node with the specified name.
-                    Static device node creation is performed by systemd-tmpfiles
-                    before systemd-udevd is started. The static nodes might not
+                    Static device node creation is performed
+                    before udevd is started. The static nodes might not
                     have a corresponding kernel device; they are used to trigger
                     automatic kernel module loading when they are accessed.</para>
                   </listitem>
@@ -708,7 +694,7 @@
                     level name is specified, the maximum log level is changed to that level. When
                     <literal>reset</literal> is set, then the previously specified log level is
                     revoked. Defaults to the log level of the main process of
-                    <command>systemd-udevd</command>.</para>
+                    <command>udevd</command>.</para>
                     <para>This may be useful when debugging events for certain devices. Note that the
                     log level is applied when the line including this rule is processed. So, for
                     debugging, it is recommended that this is specified at earlier place, e.g., the
@@ -909,15 +895,9 @@ SUBSYSTEM=="net", OPTIONS="log_level=debug"</programlisting></para>
   <refsect1>
     <title>See Also</title>
     <para><simplelist type="inline">
-      <member><citerefentry>
-        <refentrytitle>systemd-udevd.service</refentrytitle><manvolnum>8</manvolnum>
-      </citerefentry></member>
       <member><citerefentry>
         <refentrytitle>udevadm</refentrytitle><manvolnum>8</manvolnum>
       </citerefentry></member>
-      <member><citerefentry>
-        <refentrytitle>systemd.link</refentrytitle><manvolnum>5</manvolnum>
-      </citerefentry></member>
     </simplelist></para>
   </refsect1>
 </refentry>
diff --git a/man/udev_device_get_syspath.xml b/man/udev_device_get_syspath.xml
index 43269f9843..5617664446 100644
--- a/man/udev_device_get_syspath.xml
+++ b/man/udev_device_get_syspath.xml
@@ -191,7 +191,6 @@
       <member><citerefentry><refentrytitle>udev_enumerate_new</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev_monitor_new_from_netlink</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev_list_entry</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
-      <member><citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry></member>
     </simplelist></para>
   </refsect1>
 
diff --git a/man/udev_device_has_tag.xml b/man/udev_device_has_tag.xml
index 28cd5f6939..fb0ebd4019 100644
--- a/man/udev_device_has_tag.xml
+++ b/man/udev_device_has_tag.xml
@@ -177,7 +177,6 @@
       <member><citerefentry><refentrytitle>udev_enumerate_new</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev_monitor_new_from_netlink</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev_list_entry</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
-      <member><citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry></member>
     </simplelist></para>
   </refsect1>
 
diff --git a/man/udev_device_new_from_syspath.xml b/man/udev_device_new_from_syspath.xml
index 461ffb8554..4b201c51db 100644
--- a/man/udev_device_new_from_syspath.xml
+++ b/man/udev_device_new_from_syspath.xml
@@ -191,7 +191,6 @@
       <member><citerefentry><refentrytitle>udev_enumerate_new</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev_monitor_new_from_netlink</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev_list_entry</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
-      <member><citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry></member>
     </simplelist></para>
   </refsect1>
 
diff --git a/man/udev_enumerate_add_match_subsystem.xml b/man/udev_enumerate_add_match_subsystem.xml
index 24dbb3579e..54a22b5ced 100644
--- a/man/udev_enumerate_add_match_subsystem.xml
+++ b/man/udev_enumerate_add_match_subsystem.xml
@@ -142,7 +142,6 @@
       <member><citerefentry><refentrytitle>udev_enumerate_scan_devices</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev_monitor_new_from_netlink</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev_list_entry</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
-      <member><citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry></member>
     </simplelist></para>
   </refsect1>
 
diff --git a/man/udev_enumerate_new.xml b/man/udev_enumerate_new.xml
index 4941a229d3..5c6597d3c5 100644
--- a/man/udev_enumerate_new.xml
+++ b/man/udev_enumerate_new.xml
@@ -90,7 +90,6 @@
       <member><citerefentry><refentrytitle>udev_enumerate_scan_devices</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev_monitor_new_from_netlink</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev_list_entry</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
-      <member><citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry></member>
     </simplelist></para>
   </refsect1>
 
diff --git a/man/udev_enumerate_scan_devices.xml b/man/udev_enumerate_scan_devices.xml
index 47912a26f3..5541da256f 100644
--- a/man/udev_enumerate_scan_devices.xml
+++ b/man/udev_enumerate_scan_devices.xml
@@ -108,7 +108,6 @@
       <member><citerefentry><refentrytitle>udev_enumerate_add_match_subsystem</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev_monitor_new_from_netlink</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev_list_entry</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
-      <member><citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry></member>
     </simplelist></para>
   </refsect1>
 
diff --git a/man/udev_list_entry.xml b/man/udev_list_entry.xml
index 46fcde8e1a..2c702c9938 100644
--- a/man/udev_list_entry.xml
+++ b/man/udev_list_entry.xml
@@ -97,7 +97,6 @@
       <member><citerefentry><refentrytitle>udev_device_new_from_syspath</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev_enumerate_new</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev_monitor_new_from_netlink</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
-      <member><citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry></member>
     </simplelist></para>
   </refsect1>
 
diff --git a/man/udev_monitor_filter_update.xml b/man/udev_monitor_filter_update.xml
index 210b664089..6efe664c04 100644
--- a/man/udev_monitor_filter_update.xml
+++ b/man/udev_monitor_filter_update.xml
@@ -96,7 +96,6 @@
       <member><citerefentry><refentrytitle>udev_monitor_new_from_netlink</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev_monitor_receive_device</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev_list_entry</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
-      <member><citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry></member>
     </simplelist></para>
   </refsect1>
 
diff --git a/man/udev_monitor_new_from_netlink.xml b/man/udev_monitor_new_from_netlink.xml
index 9f12d746c0..b1d825fe8d 100644
--- a/man/udev_monitor_new_from_netlink.xml
+++ b/man/udev_monitor_new_from_netlink.xml
@@ -86,7 +86,6 @@
       <member><citerefentry><refentrytitle>udev_monitor_filter_update</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev_monitor_receive_device</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev_list_entry</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
-      <member><citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry></member>
     </simplelist></para>
   </refsect1>
 
diff --git a/man/udev_monitor_receive_device.xml b/man/udev_monitor_receive_device.xml
index 94d2027c60..243c89ea3d 100644
--- a/man/udev_monitor_receive_device.xml
+++ b/man/udev_monitor_receive_device.xml
@@ -112,7 +112,6 @@
       <member><citerefentry><refentrytitle>udev_monitor_new_from_netlink</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev_monitor_filter_update</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev_list_entry</refentrytitle><manvolnum>3</manvolnum></citerefentry></member>
-      <member><citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry></member>
     </simplelist></para>
   </refsect1>
 
diff --git a/man/udev_new.xml b/man/udev_new.xml
index 7be4c2211c..b9fa108f7a 100644
--- a/man/udev_new.xml
+++ b/man/udev_new.xml
@@ -78,14 +78,4 @@
     <function>udev_ref()</function>, and
     <function>udev_unref()</function> were added in version 221.</para>
   </refsect1>
-
-  <refsect1>
-    <title>See Also</title>
-
-    <para><simplelist type="inline">
-      <member><citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry></member>
-      <member><citerefentry><refentrytitle>systemd-udevd.service</refentrytitle><manvolnum>8</manvolnum></citerefentry></member>
-    </simplelist></para>
-  </refsect1>
-
 </refentry>
diff --git a/man/udevadm.xml b/man/udevadm.xml
index 2a9963cd14..cea30441d5 100644
--- a/man/udevadm.xml
+++ b/man/udevadm.xml
@@ -86,7 +86,7 @@
     <title>Description</title>
     <para><command>udevadm</command> expects a command and command
     specific options.  It controls the runtime behavior of
-    <command>systemd-udevd</command>, requests kernel events, manages
+    <command>udevd</command>, requests kernel events, manages
     the event queue, and provides simple debugging mechanisms.</para>
   </refsect1>
 
@@ -122,11 +122,7 @@
 
       <para>Positional arguments should be used to specify one or more devices. Each one may be a device name
       (in which case it must start with <filename>/dev/</filename>), a sys path (in which case it must start
-      with <filename>/sys/</filename>), a device ID (such as <literal>n1</literal>, <literal>c5:1</literal>,
-      or <literal>b259:1</literal>, see
-      <citerefentry><refentrytitle>sd_device_get_device_id</refentrytitle><manvolnum>3</manvolnum></citerefentry>).
-      or a systemd device unit name (in which case it must end with <literal>.device</literal>, see
-      <citerefentry><refentrytitle>systemd.device</refentrytitle><manvolnum>5</manvolnum></citerefentry>).
+      with <filename>/sys/</filename>).
       </para>
 
       <variablelist>
@@ -589,10 +585,10 @@
           <term><option>--initialized-nomatch</option></term>
           <listitem>
             <para>When <option>--initialized-match</option> is specified, trigger events for devices
-            that are already initialized by <command>systemd-udevd</command>, and skip devices that
+            that are already initialized by <command>udevd</command>, and skip devices that
             are not initialized yet.</para>
             <para>When <option>--initialized-nomatch</option> is specified, trigger events for devices
-            that are not initialized by <command>systemd-udevd</command> yet, and skip devices that
+            that are not initialized by <command>udevd</command> yet, and skip devices that
             are already initialized.</para>
             <para>Typically, it is essential that applications which intend to use such a match, make
             sure a suitable udev rule is installed that sets at least one property on devices that
@@ -649,7 +645,7 @@
         <varlistentry>
           <term><option>--wait-daemon[=<replaceable>SECONDS</replaceable>]</option></term>
           <listitem>
-            <para>Before triggering uevents, wait for systemd-udevd daemon to be initialized.
+            <para>Before triggering uevents, wait for udevd daemon to be initialized.
             Optionally takes timeout value. Default timeout is 5 seconds. This is equivalent to invoking
             <command>udevadm control --ping</command> before <command>udevadm trigger</command>.</para>
 
@@ -692,10 +688,6 @@
           </listitem>
         </varlistentry>
       </variablelist>
-
-      <para>See
-      <citerefentry><refentrytitle>systemd-udev-settle.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>
-      for more information.</para>
     </refsect2>
 
     <refsect2>
@@ -708,12 +700,8 @@
           <term><option>-e</option></term>
           <term><option>--exit</option></term>
           <listitem>
-            <para>Signal and wait for systemd-udevd to exit. No option except for
+            <para>Signal and wait for udevd to exit. No option except for
             <option>--timeout</option> can be specified after this option.
-            Note that <filename>systemd-udevd.service</filename> contains
-            <option>Restart=always</option> and so as a result, this option restarts systemd-udevd.
-            If you want to stop <filename>systemd-udevd.service</filename>, please use the following:
-            <programlisting>systemctl stop systemd-udevd-control.socket systemd-udevd-kernel.socket systemd-udevd-varlink.socket systemd-udevd.service</programlisting>
             </para>
           </listitem>
         </varlistentry>
@@ -722,7 +710,7 @@
           <term><option>--log-level=<replaceable>value</replaceable></option></term>
           <listitem>
             <para>Set the internal log level of
-            <filename>systemd-udevd</filename>. Valid values are the
+            <filename>udevd</filename>. Valid values are the
             numerical syslog priorities or their textual
             representations: <option>emerg</option>,
             <option>alert</option>, <option>crit</option>,
@@ -735,7 +723,7 @@
           <term><option>-s</option></term>
           <term><option>--stop-exec-queue</option></term>
           <listitem>
-            <para>Signal systemd-udevd to stop executing new events. Incoming events
+            <para>Signal udevd to stop executing new events. Incoming events
             will be queued.</para>
           </listitem>
         </varlistentry>
@@ -743,14 +731,14 @@
           <term><option>-S</option></term>
           <term><option>--start-exec-queue</option></term>
           <listitem>
-            <para>Signal systemd-udevd to enable the execution of events.</para>
+            <para>Signal udevd to enable the execution of events.</para>
           </listitem>
         </varlistentry>
         <varlistentry>
           <term><option>-R</option></term>
           <term><option>--reload</option></term>
           <listitem>
-            <para>Signal systemd-udevd to reload the rules files and other databases like the kernel
+            <para>Signal udevd to reload the rules files and other databases like the kernel
             module index. Reloading rules and databases does not apply any changes to already
             existing devices; the new configuration will only be applied to new events.</para>
           </listitem>
@@ -766,15 +754,15 @@
           <term><option>-m</option></term>
           <term><option>--children-max=<replaceable>value</replaceable></option></term>
           <listitem>
-            <para>Set the maximum number of events, systemd-udevd will handle at the same time. When 0 is
+            <para>Set the maximum number of events, udevd will handle at the same time. When 0 is
             specified, then the maximum is determined based on the system resources.</para>
           </listitem>
         </varlistentry>
         <varlistentry>
           <term><option>--ping</option></term>
           <listitem>
-            <para>Send a ping message to systemd-udevd and wait for the reply. This may be useful to check that
-            systemd-udevd daemon is running.</para>
+            <para>Send a ping message to udevd and wait for the reply. This may be useful to check that
+            udevd daemon is running.</para>
 
             <xi:include href="version-info.xml" xpointer="v241"/>
           </listitem>
@@ -782,7 +770,7 @@
         <varlistentry>
           <term><option>--trace=<replaceable>BOOL</replaceable></option></term>
           <listitem>
-            <para>Enable/disable trace logging in <command>systemd-udevd</command>.</para>
+            <para>Enable/disable trace logging in <command>udevd</command>.</para>
 
             <xi:include href="version-info.xml" xpointer="v258"/>
           </listitem>
@@ -801,7 +789,7 @@
           <term><option>-t</option></term>
           <term><option>--timeout=<replaceable>seconds</replaceable></option></term>
           <listitem>
-            <para>The maximum number of seconds to wait for a reply from systemd-udevd.</para>
+            <para>The maximum number of seconds to wait for a reply from udevd.</para>
           </listitem>
         </varlistentry>
 
@@ -838,9 +826,9 @@
             </variablelist>
 
             <para>Note, this <emphasis>does not</emphasis> imply <option>--reload</option> option. So, if
-            <command>systemd-udevd</command> is already running, please consider to also specify
+            <command>udevd</command> is already running, please consider to also specify
             <option>-reload</option> to make the copied udev rules files used by
-            <command>systemd-udevd</command>.</para>
+            <command>udevd</command>.</para>
 
             <xi:include href="version-info.xml" xpointer="v256"/>
           </listitem>
@@ -945,7 +933,7 @@
             multiple times. It may be useful for debugging udev rules by copying some udev rules files to a
             temporary directory, editing them, and specifying the directory with this option. Files found in
             the specified directories have a higher priority than rules files in the default
-            <filename>udev/rules.d</filename> directories used by <command>systemd-udevd</command>. See
+            <filename>udev/rules.d</filename> directories used by <command>udevd</command>. See
             <citerefentry><refentrytitle>udev</refentrytitle><manvolnum>7</manvolnum></citerefentry> for more
             details about the search paths.</para>
 
@@ -1006,9 +994,9 @@
       absolute path to a udev rules file or a directory that contains rules files, or a file name of udev
       rules file (e.g. <filename>99-systemd.rules</filename>). If a file name is specified, the file will be
       searched in the <filename>udev/rules.d</filename> directories that are processed by
-      <command>systemd-udevd</command>, and searched in the current working directory if not found. If no
+      <command>udevd</command>, and searched in the current working directory if not found. If no
       files are specified, the udev rules are read from the files located in the same
-      <filename>udev/rules.d</filename> directories that are processed by <command>systemd-udevd</command>.
+      <filename>udev/rules.d</filename> directories that are processed by <command>udevd</command>.
       See <citerefentry><refentrytitle>udev</refentrytitle><manvolnum>7</manvolnum></citerefentry> for more
       details about the search paths.</para>
 
@@ -1024,7 +1012,7 @@
             <para>Specify when <command>udevadm</command> should resolve names of users and groups specified
             in udev rules. When set to <constant>early</constant> (the default), names will be resolved when
             the rules are parsed. When set to <constant>late</constant>, names will not be verified, as
-            <command>systemd-udevd</command> resolves names during each event being processed. When set to
+            <command>udevd</command> resolves names during each event being processed. When set to
             <constant>never</constant>, names will never be resolved and relevant rules will be ignored.
             </para>
 
@@ -1078,9 +1066,9 @@
       absolute path to a udev rules file or a directory that contains rules files, or a file name of udev
       rules file (e.g. <filename>99-systemd.rules</filename>). If a file name is specified, the file will be
       searched in the <filename>udev/rules.d</filename> directories that are processed by
-      <command>systemd-udevd</command>, and searched in the current working directory if not found. If no
+      <command>udevd</command>, and searched in the current working directory if not found. If no
       files are specified, the udev rules are read from the files located in the same
-      <filename>udev/rules.d</filename> directories that are processed by <command>systemd-udevd</command>.
+      <filename>udev/rules.d</filename> directories that are processed by <command>udevd</command>.
       See <citerefentry><refentrytitle>udev</refentrytitle><manvolnum>7</manvolnum></citerefentry> for more
       details about the search paths.</para>
 
@@ -1129,12 +1117,12 @@
       </title>
 
       <para>Wait for devices or device symlinks being created and initialized by
-      <command>systemd-udevd</command>. Each device path must start with
+      <command>udevd</command>. Each device path must start with
       <literal>/dev/</literal> or <literal>/sys/</literal>, e.g. <literal>/dev/sda</literal>,
       <literal>/dev/disk/by-path/pci-0000:3c:00.0-nvme-1-part1</literal>,
       <literal>/sys/devices/pci0000:00/0000:00:1f.6/net/eth0</literal>, or
       <literal>/sys/class/net/eth0</literal>. This can take multiple devices. This may be useful for
-      waiting for devices being processed by <command>systemd-udevd</command> after e.g. partitioning
+      waiting for devices being processed by <command>udevd</command> after e.g. partitioning
       or formatting the devices.</para>
 
       <variablelist>
@@ -1152,7 +1140,7 @@
         <varlistentry>
           <term><option>--initialized=<replaceable>BOOL</replaceable></option></term>
           <listitem>
-            <para>Check if <command>systemd-udevd</command> initialized devices. Defaults to true. When
+            <para>Check if <command>udevd</command> initialized devices. Defaults to true. When
             false, the command only checks if the specified devices exist. Set false to this setting if
             there is no udev rules for the specified devices, as the devices will never be considered
             as initialized in that case. See Initialized Devices section below for more details.</para>
@@ -1175,7 +1163,7 @@
           <term><option>--settle</option></term>
           <listitem>
             <para>When specified, also watches the udev event queue, and wait for all queued events
-            being processed by <command>systemd-udevd</command>.</para>
+            being processed by <command>udevd</command>.</para>
 
             <xi:include href="version-info.xml" xpointer="v251"/>
           </listitem>
@@ -1195,14 +1183,6 @@
       are automatically released and its return value is propagated as exit code of <command>udevadm
       lock</command>.</para>
 
-      <para>This tool is in particular useful to ensure that
-      <citerefentry><refentrytitle>systemd-udevd.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>
-      does not probe a block device while changes are made to it, for example partitions created or file
-      systems formatted. Note that many tools that interface with block devices natively support taking
-      relevant locks, see for example
-      <citerefentry project='man-pages'><refentrytitle>sfdisk</refentrytitle><manvolnum>8</manvolnum></citerefentry>'s
-      <option>--lock</option> switch.</para>
-
       <para>The command expects at least one block device specified via <option>--device=</option> or
       <option>--backing=</option>, and a command line to execute as arguments.</para>
 
@@ -1247,8 +1227,7 @@
 
           <listitem>
             <para>Specifies how long to wait at most until all locks can be taken. Takes a value in
-            seconds, or in the usual supported time units, see
-            <citerefentry><refentrytitle>systemd.time</refentrytitle><manvolnum>7</manvolnum></citerefentry>. If
+            seconds, or in the usual supported time units. If
             specified as zero the lock is attempted and if not successful the invocation will immediately
             fail. If passed as <literal>infinity</literal> (the default) the invocation will wait indefinitely
             until the lock can be acquired. If the lock cannot be taken in the specified time the specified
@@ -1297,7 +1276,7 @@
       <title>Format a File System</title>
 
       <para>Take a lock on the backing block device while creating a file system, to ensure that
-      <command>systemd-udevd</command> does not probe or announce the new superblock before it is
+      <command>udevd</command> does not probe or announce the new superblock before it is
       comprehensively written:</para>
 
       <programlisting># udevadm lock --device=/dev/sda1 mkfs.ext4 /dev/sda1</programlisting>
@@ -1315,7 +1294,7 @@
       <title>Copy in a File System</title>
 
       <para>Take a lock on the backing block device while copying in a prepared file system image, to ensure
-      that <command>systemd-udevd</command> does not probe or announce the new superblock before it is fully
+      that <command>udevd</command> does not probe or announce the new superblock before it is fully
       written:</para>
 
       <programlisting># udevadm lock -d /dev/sda1 dd if=fs.raw of=/dev/sda1</programlisting>
@@ -1356,7 +1335,6 @@ $ udevadm test /dev/input/event3 --json=pretty 2>/dev/null | jq .node.symlinks
     <title>See Also</title>
     <para><simplelist type="inline">
       <member><citerefentry><refentrytitle>udev</refentrytitle><manvolnum>7</manvolnum></citerefentry></member>
-      <member><citerefentry><refentrytitle>systemd-udevd.service</refentrytitle><manvolnum>8</manvolnum></citerefentry></member>
       <member><citerefentry><refentrytitle>udev.conf</refentrytitle><manvolnum>5</manvolnum></citerefentry></member>
     </simplelist></para>
   </refsect1>
diff --git a/meson.build b/meson.build
index 438ce4f5f5..e1eacfe065 100644
--- a/meson.build
+++ b/meson.build
@@ -765,7 +765,7 @@ if time_epoch <= 0
         endif
         if time_epoch == ''
                 NEWS = files('NEWS')
-                time_epoch = run_command(stat, '-c', '%Y', NEWS,
+                time_epoch = run_command(stat, '-c', '%m', NEWS,
                                          check : false)
                 if time_epoch.returncode() != 0
                         # If the above fails, maybe the stat(1) uses BSD-style syntax
@@ -1999,126 +1999,9 @@ subdir('src/libsystemd-network')
 # hwdb requires 'udev_link_with' and 'udev_rpath'
 subdir('src/udev')
 
-subdir('src/ac-power')
-subdir('src/analyze')
-subdir('src/ask-password')
-subdir('src/backlight')
-subdir('src/battery-check')
-subdir('src/binfmt')
-subdir('src/bless-boot')
-subdir('src/boot')
-subdir('src/bootctl')
-subdir('src/busctl')
-subdir('src/cgls')
-subdir('src/cgtop')
-subdir('src/coredump')
-subdir('src/creds')
-subdir('src/cryptenroll')
-subdir('src/cryptsetup')
-subdir('src/debug-generator')
-subdir('src/delta')
-subdir('src/detect-virt')
-subdir('src/dissect')
-subdir('src/environment-d-generator')
-subdir('src/escape')
-subdir('src/factory-reset')
-subdir('src/firstboot')
-subdir('src/fsck')
-subdir('src/fstab-generator')
-subdir('src/getty-generator')
-subdir('src/gpt-auto-generator')
-subdir('src/growfs')
-subdir('src/hibernate-resume')
-subdir('src/home')
-subdir('src/hostname')
 subdir('src/hwdb')
-subdir('src/id128')
-subdir('src/import')
-subdir('src/imds') # Note, we are not alphabetically here, since we want to use a variable from src/import/ here
-subdir('src/integritysetup')
-subdir('src/journal')
-subdir('src/journal-remote')
-subdir('src/kernel-install')
-subdir('src/keyutil')
-subdir('src/locale')
-subdir('src/login')
-subdir('src/machine')
-subdir('src/machine-id-setup')
-subdir('src/measure')
-subdir('src/modules-load')
-subdir('src/mount')
-subdir('src/mountfsd')
-subdir('src/mstack')
-subdir('src/mute-console')
-subdir('src/network')
-subdir('src/notify')
-subdir('src/nspawn')
-subdir('src/nsresourced')
-subdir('src/nss-myhostname')
-subdir('src/nss-mymachines')
-subdir('src/nss-resolve')
-subdir('src/nss-systemd')
-subdir('src/oom')
-subdir('src/path')
-subdir('src/pcrextend')
-subdir('src/pcrlock')
-subdir('src/portable')
-subdir('src/pstore')
-subdir('src/ptyfwd')
-subdir('src/quotacheck')
-subdir('src/random-seed')
-subdir('src/remount-fs')
-subdir('src/repart')
-subdir('src/report')
-subdir('src/reply-password')
-subdir('src/resolve')
-subdir('src/rfkill')
-subdir('src/rpm')
-subdir('src/run')
-subdir('src/run-generator')
-subdir('src/sbsign')
-subdir('src/shutdown')
-subdir('src/sleep')
-subdir('src/socket-activate')
-subdir('src/socket-proxy')
-subdir('src/ssh-generator')
-subdir('src/stdio-bridge')
-subdir('src/storage')
-subdir('src/storagetm')
-subdir('src/sulogin-shell')
-subdir('src/sysctl')
-subdir('src/sysext')
-subdir('src/sysinstall')
-subdir('src/system-update-generator')
-subdir('src/systemctl')
-subdir('src/sysupdate')
-subdir('src/sysusers')
-subdir('src/timedate')
-subdir('src/timesync')
-subdir('src/tmpfiles')
-subdir('src/tpm2-setup')
-subdir('src/tty-ask-password-agent')
-subdir('src/update-done')
-subdir('src/update-utmp')
-subdir('src/user-sessions')
-subdir('src/userdb')
-subdir('src/validatefs')
-subdir('src/varlinkctl')
-subdir('src/vconsole')
-subdir('src/veritysetup')
-subdir('src/vmspawn')
-subdir('src/volatile-root')
-subdir('src/vpick')
-subdir('src/xdg-autostart-generator')
-
-subdir('src/systemd')
 
 subdir('src/test')
-subdir('src/fuzz')
-subdir('src/ukify/test')  # needs to be last for test_env variable
-subdir('test/fuzz')
-
-subdir('mime')
 
 alias_target('devel', libsystemd_pc, libudev_pc, systemd_pc, udev_pc)
 
@@ -2132,10 +2015,6 @@ foreach test : libsystemd_tests
         executables += test_template + test
 endforeach
 
-foreach fuzzer : simple_fuzzers
-        executables += fuzz_template + { 'sources' : [fuzzer] }
-endforeach
-
 foreach dict : executables
         name = dict.get('name', '')
         if name == ''
@@ -2200,17 +2079,6 @@ foreach dict : executables
                 endforeach
         endif
 
-        if is_fuzz
-                include_directories += include_directories('src/fuzz')
-                foreach key, val : fuzz_additional_kwargs
-                        if key == 'sources'
-                                exe_sources += val
-                        else
-                                kwargs += { key : [ kwargs.get(key, []), val ] }
-                        endif
-                endforeach
-        endif
-
         build_by_default = dict.get('build_by_default',
                                     have_standalone_binaries or not is_standalone)
 
@@ -2276,45 +2144,8 @@ foreach dict : executables
                              is_parallel : dict.get('parallel', true))
                 endif
         endif
-
-        if is_fuzz
-                fuzzer_exes += exe
-
-                if want_tests != 'false' and want_fuzz_tests
-                        fuzz_ins = fuzz_regression_tests.get(name, {})
-                        foreach directive : fuzz_ins.get('directives', [])
-                                tt = '@0@_@1@'.format(name, fs.name(directive.full_path()))
-                                if tt.substring(45) != ''
-                                        error('Directive sample name is too long:', directive.full_path())
-                                endif
-
-                                test(tt,
-                                     exe,
-                                     suite : 'fuzz',
-                                     args : directive.full_path(),
-                                     env : ['UBSAN_OPTIONS=print_stacktrace=1:print_summary=1:halt_on_error=1'],
-                                     timeout : 60,
-                                     depends : directive)
-                        endforeach
-                        foreach file : fuzz_ins.get('files', [])
-                                tt = '@0@_@1@'.format(name, fs.name(file))
-                                if tt.substring(45) != ''
-                                        error('Fuzz sample name is too long:', fs.name(file))
-                                endif
-
-                                test(tt,
-                                     exe,
-                                     suite : 'fuzz',
-                                     env : ['UBSAN_OPTIONS=print_stacktrace=1:print_summary=1:halt_on_error=1'],
-                                     timeout : 60,
-                                     args : file)
-                        endforeach
-                endif
-        endif
 endforeach
 
-alias_target('fuzzers', fuzzer_exes)
-
 #####################################################################
 
 test_dlopen = executables_by_name.get('test-dlopen')
@@ -2465,44 +2296,13 @@ endif
 
 ############################################################
 
-subdir('rules.d')
-subdir('test')
-
 #####################################################################
 
-subdir('docs/var-log')
+subdir('rules.d')
 subdir('hwdb.d')
 subdir('man')
-subdir('modprobe.d')
-subdir('network')
-subdir('presets')
-subdir('profile.d')
 subdir('shell-completion/bash')
 subdir('shell-completion/zsh')
-subdir('sysctl.d')
-subdir('sysusers.d')
-subdir('tmpfiles.d')
-subdir('units')
-
-install_subdir('factory/etc',
-               install_dir : factorydir)
-subdir('factory/templates')
-
-if install_sysconfdir
-        install_data('xorg/50-systemd-user.sh',
-                     install_dir : xinitrcdir)
-endif
-install_data('LICENSE.GPL2',
-             'LICENSE.LGPL2.1',
-             'NEWS',
-             'README',
-             'docs/ENVIRONMENT.md',
-             'docs/TRANSIENT-SETTINGS.md',
-             'docs/UIDS-GIDS.md',
-             install_dir : docdir)
-
-install_subdir('LICENSES',
-               install_dir : docdir)
 
 #####################################################################
 
@@ -2595,53 +2395,6 @@ endif
 
 alias_target('gensources', generated_sources)
 
-clang_tidy = find_program('clang-tidy', required : false)
-if meson.version().version_compare('>=1.4.0')
-        uniq = {}
-
-        foreach source : sources
-                if uniq.has_key(source.full_path())
-                        continue
-                endif
-                uniq += {source.full_path(): source}
-        endforeach
-
-        # TODO: Use uniq.values() when we can rely on meson 1.10.0.
-        sources = []
-        foreach path, file : uniq
-                sources += [file]
-        endforeach
-
-        foreach source : sources
-                if systemd_headers.contains(source)
-                        continue
-                endif
-
-                if not source.full_path().endswith('.c') and not source.full_path().endswith('.h')
-                        continue
-                endif
-
-                inputs = [source]
-
-                header = source.full_path().replace('.c', '.h')
-                if fs.exists(header) and header != source.full_path() and header != libudev_h_path
-                        inputs += header
-                endif
-
-                foreach input : inputs
-                        if clang_tidy.found()
-                                test(
-                                        'clang-tidy-@0@'.format(fs.name(input)),
-                                        clang_tidy,
-                                        args : ['-p', meson.project_build_root(), input],
-                                        suite : 'clang-tidy',
-                                        depends : generated_sources,
-                                )
-                        endif
-                endforeach
-        endforeach
-endif
-
 spatch = find_program('spatch', required : false)
 if spatch.found()
         coccinelle_exclude = [
@@ -2691,7 +2444,6 @@ run_target(
 
 alias_target('man', man)
 alias_target('html', html)
-alias_target('update-dbus-docs', update_dbus_docs)
 alias_target('update-man-rules', update_man_rules)
 
 if not meson.is_cross_build()
@@ -2717,18 +2469,11 @@ udev_targets = []
 foreach bin : udev_binaries
         udev_targets += executables_by_name.get(bin, [])
 endforeach
-alias_target('udev', buildable_rules, udev_targets, udev_units)
+alias_target('udev', buildable_rules, udev_targets)
 
 if conf.get('ENABLE_HWDB') == 1
-        alias_target('hwdb', auto_suspend_rules, executables_by_name.get('systemd-hwdb'), hwdb_units)
-endif
-
-alt_time_epoch = run_command('date', '-Is', '-u', '-d', f'@@time_epoch@', check : false)
-if alt_time_epoch.returncode() != 0
-        # If the above fails, maybe the date(1) uses BSD-style syntax
-        alt_time_epoch = run_command('date', '-Iseconds', '-u', '-r', f'@time_epoch@', check : true)
+        alias_target('hwdb', auto_suspend_rules, executables_by_name.get('udev-hwdb'))
 endif
-alt_time_epoch = alt_time_epoch.stdout().strip()
 
 summary({
         'split bin-sbin'                            : split_bin,
@@ -2785,7 +2530,6 @@ summary({
         'default journal storage mode'              : conf.get('JOURNAL_STORAGE_DEFAULT'),
         'default user $PATH'                        : default_user_path != '' ? default_user_path : '(same as system services)',
         'systemd service watchdog'                  : service_watchdog == '' ? 'disabled' : service_watchdog,
-        'time epoch'                                : f'@time_epoch@ (@alt_time_epoch@)',
         'TPM2 nvpcr base'                           : run_command(sh, '-c', 'printf 0x%x @0@'.format(get_option('tpm2-nvpcr-base')), check : true).stdout(),
         'IMDS networking'                           : get_option('imds-network'),
 })
@@ -2916,7 +2660,6 @@ foreach tuple : [
         ['debug siphash'],
         ['trace logging',          conf.get('LOG_TRACE') == 1],
         ['slow tests',             want_slow_tests],
-        ['fuzz tests',             want_fuzz_tests],
         ['install tests',          install_tests],
         ['link-udev-shared',       get_option('link-udev-shared')],
         ['link-systemctl-shared',  get_option('link-systemctl-shared')],
diff --git a/rules.d/README b/rules.d/README
index 08edb4da55..58871daa77 100644
--- a/rules.d/README
+++ b/rules.d/README
@@ -1,8 +1,5 @@
-Files in this directory contain configuration for systemd-udevd.service, a
+Files in this directory contain configuration for udevd, a
 daemon that manages symlinks to device nodes, permissions of devices nodes,
 emits device events for userspace, and renames network interfaces.
 
-See man:udev(7) for an overview of the configuration file format, and
-man:systemd-udevd.service(8) for a description of service itself.
-
-Use 'systemd-analyze cat-config udev/rules.d' to display the effective config.
+See man:udev(7) for an overview of the configuration file format.
diff --git a/rules.d/meson.build b/rules.d/meson.build
index aa469ef7a7..e936f5b11f 100644
--- a/rules.d/meson.build
+++ b/rules.d/meson.build
@@ -45,9 +45,6 @@ rules = [
 
         [files('70-memory.rules'),
          conf.get('HAVE_DMI') == 1],
-
-        [files('70-power-switch.rules'),
-         conf.get('ENABLE_LOGIND') == 1]
 ]
 
 all_rules = []
@@ -67,11 +64,6 @@ rules_in = [
         ['50-udev-default.rules'],
         ['60-persistent-storage.rules'],
         ['64-btrfs.rules'],
-        ['99-systemd.rules'],
-
-        ['70-uaccess.rules',   conf.get('ENABLE_LOGIND') == 1 and conf.get('HAVE_ACL') == 1],
-        ['71-seat.rules',      conf.get('ENABLE_LOGIND') == 1],
-        ['73-seat-late.rules', conf.get('ENABLE_LOGIND') == 1],
 
         ['90-vconsole.rules',  conf.get('ENABLE_VCONSOLE') == 1],
 ]
diff --git a/shell-completion/bash/meson.build b/shell-completion/bash/meson.build
index cddf742059..1f240bb0cf 100644
--- a/shell-completion/bash/meson.build
+++ b/shell-completion/bash/meson.build
@@ -16,51 +16,12 @@ custom_target(
         input : 'systemctl.in',
         output : 'systemctl',
         command : [jinja2_cmdline, '@INPUT@', '@OUTPUT@'],
-        install : true,
+        install : false,
         install_dir : bashcompletiondir)
 
 foreach item : [
-        ['bootctl',             ''],
-        ['busctl',              ''],
-        ['coredumpctl',         'ENABLE_COREDUMP'],
-        ['homectl',             'ENABLE_HOMED'],
-        ['hostnamectl',         'ENABLE_HOSTNAMED'],
-        ['importctl',           'ENABLE_IMPORTD'],
-        ['journalctl',          ''],
-        ['kernel-install',      'ENABLE_KERNEL_INSTALL'],
-        ['localectl',           'ENABLE_LOCALED'],
-        ['loginctl',            'ENABLE_LOGIND'],
-        ['machinectl',          'ENABLE_MACHINED'],
-        ['networkctl',          'ENABLE_NETWORKD'],
-        ['oomctl',              'ENABLE_OOMD'],
-        ['portablectl',         'ENABLE_PORTABLED'],
-        ['resolvectl',          'ENABLE_RESOLVE'],
-        ['run0',                ''],
-        ['storagectl',          ''],
-        ['systemd-analyze',     ''],
-        ['systemd-cat',         ''],
-        ['systemd-cgls',        ''],
-        ['systemd-cgtop',       ''],
-        ['systemd-confext',     'ENABLE_SYSEXT'],
-        ['systemd-creds',       ''],
-        ['systemd-cryptenroll', 'HAVE_LIBCRYPTSETUP'],
-        ['systemd-delta',       ''],
-        ['systemd-detect-virt', ''],
-        ['systemd-dissect',     'HAVE_BLKID'],
-        ['systemd-hwdb',        'ENABLE_HWDB'],
-        ['systemd-id128',       ''],
-        ['systemd-nspawn',      'ENABLE_NSPAWN'],
-        ['systemd-path',        ''],
-        ['systemd-resolve',     'ENABLE_RESOLVE'],
-        ['systemd-run',         ''],
-        ['systemd-sysext',      'ENABLE_SYSEXT'],
-        ['systemd-sysinstall',  'ENABLE_SYSINSTALL'],
-        ['systemd-vmspawn',     'ENABLE_VMSPAWN'],
-        ['systemd-vpick',       ''],
-        ['timedatectl',         'ENABLE_TIMEDATED'],
+        ['udev-hwdb',        'ENABLE_HWDB'],
         ['udevadm',             ''],
-        ['userdbctl',           'ENABLE_USERDB'],
-        ['varlinkctl',          ''],
 ]
 
         if item[1] == '' or conf.get(item[1]) == 1
diff --git a/shell-completion/bash/systemd-hwdb b/shell-completion/bash/udev-hwdb
similarity index 94%
rename from shell-completion/bash/systemd-hwdb
rename to shell-completion/bash/udev-hwdb
index a401bbf1c7..9085002a7c 100644
--- a/shell-completion/bash/systemd-hwdb
+++ b/shell-completion/bash/udev-hwdb
@@ -1,5 +1,5 @@
 # shellcheck shell=bash
-# systemd-hwdb(8) completion                    -*- shell-script -*-
+# udev-hwdb(8) completion                    -*- shell-script -*-
 # SPDX-License-Identifier: LGPL-2.1-or-later
 #
 # This file is part of systemd.
@@ -24,7 +24,7 @@ __contains_word() {
     done
 }
 
-_systemd_hwdb() {
+_udev_hwdb() {
     local cur=${COMP_WORDS[COMP_CWORD]} prev=${COMP_WORDS[COMP_CWORD-1]} words cword
     local i verb comps
 
@@ -73,4 +73,4 @@ _systemd_hwdb() {
     return 0
 }
 
-complete -F _systemd_hwdb systemd-hwdb
+complete -F _udev_hwdb udev-hwdb
diff --git a/shell-completion/zsh/_systemd-hwdb b/shell-completion/zsh/_udev-hwdb
similarity index 91%
rename from shell-completion/zsh/_systemd-hwdb
rename to shell-completion/zsh/_udev-hwdb
index 92238d63ad..d9771253bc 100644
--- a/shell-completion/zsh/_systemd-hwdb
+++ b/shell-completion/zsh/_udev-hwdb
@@ -1,4 +1,4 @@
-#compdef systemd-hwdb
+#compdef udev-hwdb
 # SPDX-License-Identifier: LGPL-2.1-or-later
 
 local context state state_descr line
@@ -24,7 +24,7 @@ _arguments -s -A '-*' "$opt_common[@]" \
 
 case $state in
     command)
-        _describe -t command 'systemd-hwdb command' hwdb_commands && ret=0
+        _describe -t command 'udev-hwdb command' hwdb_commands && ret=0
         ;;
     option-or-argument)
         case $words[1] in
diff --git a/shell-completion/zsh/_udevadm b/shell-completion/zsh/_udevadm
index 7e717d30a3..dff1599f90 100644
--- a/shell-completion/zsh/_udevadm
+++ b/shell-completion/zsh/_udevadm
@@ -61,16 +61,16 @@ _udevadm_control(){
     _arguments \
         '(-)'{-h,--help}'[Show help]' \
         '(-)'{-V,--version}'[Show package version]' \
-        '(-e --exit)'{-e,--exit}'[Signal and wait for systemd-udevd to exit.]' \
-        '(-l --log-level)'{-l,--log-level=}'[Set the internal log level of systemd-udevd.]:LEVEL:(err info debug)' \
-        '(-s --stop-exec-queue)'{-s,--stop-exec-queue}'[Signal systemd-udevd to stop executing new events. Incoming events will be queued.]' \
-        '(-S --start-exec-queue)'{-S,--start-exec-queue}'[Signal systemd-udevd to enable the execution of events.]' \
-        '(-R --reload)'{-R,--reload}'[Signal systemd-udevd to reload the rules files and other databases like the kernel module index.]' \
+        '(-e --exit)'{-e,--exit}'[Signal and wait for udevd to exit.]' \
+        '(-l --log-level)'{-l,--log-level=}'[Set the internal log level of udevd.]:LEVEL:(err info debug)' \
+        '(-s --stop-exec-queue)'{-s,--stop-exec-queue}'[Signal udevd to stop executing new events. Incoming events will be queued.]' \
+        '(-S --start-exec-queue)'{-S,--start-exec-queue}'[Signal udevd to enable the execution of events.]' \
+        '(-R --reload)'{-R,--reload}'[Signal udevd to reload the rules files and other databases like the kernel module index.]' \
         '(-p --property)'{-p,--property=}'[Set a global property for all events.]:KEY=VALUE' \
         '(-m --children-max=)'{-m,--children-max=}'[Set the maximum number of events.]:N' \
         '--trace=[Enable/disable trace logging.]:BOOL' \
         '--revert[Revert previously set configurations.]' \
-        '(-t --timeout=)'{-t,--timeout=}'[The maximum number of seconds to wait for a reply from systemd-udevd.]:SECONDS'
+        '(-t --timeout=)'{-t,--timeout=}'[The maximum number of seconds to wait for a reply from udevd.]:SECONDS'
 }
 
 (( $+functions[_udevadm_monitor] )) ||
@@ -148,7 +148,7 @@ _udevadm_wait(){
         '(- *)'{-h,--help}'[Print help]' \
         '(- *)'{-V,--version}'[Print version of the program]' \
         '--timeout=[Maximum number of seconds to wait for the devices being created.]:SEC' \
-        '--initialized=[Wait for devices being initialized by systemd-udevd.]:boolean:(yes no)' \
+        '--initialized=[Wait for devices being initialized by udevd.]:boolean:(yes no)' \
         '--removed[Wait for devices being removed.]' \
         '--settle[Also wait for udev queue being empty.]' \
         '*::devpath:_files -W /dev'
diff --git a/shell-completion/zsh/meson.build b/shell-completion/zsh/meson.build
index f10ba7be61..f214f4d81d 100644
--- a/shell-completion/zsh/meson.build
+++ b/shell-completion/zsh/meson.build
@@ -11,44 +11,12 @@ custom_target(
         input : '_systemctl.in',
         output : '_systemctl',
         command : [jinja2_cmdline, '@INPUT@', '@OUTPUT@'],
-        install : true,
+        install : false,
         install_dir : zshcompletiondir)
 
 foreach item : [
-        ['_bootctl',                  ''],
-        ['_busctl',                   ''],
-        ['_coredumpctl',              'ENABLE_COREDUMP'],
-        ['_hostnamectl',              'ENABLE_HOSTNAMED'],
-        ['_journalctl',               ''],
-        ['_kernel-install',           'ENABLE_KERNEL_INSTALL'],
-        ['_localectl',                'ENABLE_LOCALED'],
-        ['_loginctl',                 'ENABLE_LOGIND'],
-        ['_machinectl',               'ENABLE_MACHINED'],
-        ['_networkctl',               'ENABLE_NETWORKD'],
-        ['_oomctl',                   'ENABLE_OOMD'],
-        ['_resolvectl',               'ENABLE_RESOLVE'],
-        ['_run0',                     ''],
-        ['_sd_bus_address',           ''],
-        ['_sd_hosts_or_user_at_host', ''],
-        ['_sd_machines',              'ENABLE_MACHINED'],
-        ['_sd_outputmodes',           ''],
-        ['_sd_unit_files',            ''],
-        ['_storagectl',               ''],
-        ['_systemd',                  ''],
-        ['_systemd-analyze',          ''],
-        ['_systemd-delta',            ''],
-        ['_systemd-hwdb',             'ENABLE_HWDB'],
-        ['_systemd-id128',            ''],
-        ['_systemd-inhibit',          'ENABLE_LOGIND'],
-        ['_systemd-nspawn',           ''],
-        ['_systemd-path',             ''],
-        ['_systemd-run',              ''],
-        ['_systemd-sysinstall',       'ENABLE_SYSINSTALL'],
-        ['_systemd-tmpfiles',         'ENABLE_TMPFILES'],
-        ['_timedatectl',              'ENABLE_TIMEDATED'],
+        ['_udev-hwdb',             'ENABLE_HWDB'],
         ['_udevadm',                  ''],
-        ['_userdbctl',                ''],
-        ['_varlinkctl',               ''],
 ]
 
         if item[1] == '' or conf.get(item[1]) == 1
diff --git a/src/core/meson.build b/src/core/meson.build
index af50a2a4fa..14a0e1f330 100644
--- a/src/core/meson.build
+++ b/src/core/meson.build
@@ -294,21 +294,6 @@ install_emptydir(systemgeneratordir)
 install_emptydir(userenvgeneratordir)
 install_emptydir(usergeneratordir)
 
-if install_sysconfdir
-        install_emptydir(pkgsysconfdir / 'system')
-        install_emptydir(pkgsysconfdir / 'user')
-
-        assert(sysconfdir / 'systemd' == pkgsysconfdir)
-        install_symlink('user',
-                        pointing_to : '../../systemd/user',
-                        install_dir : sysconfdir / 'xdg/systemd')
-endif
-
-assert(fs.parent(sbindir) / 'lib/systemd' == libexecdir)
-install_symlink('init',
-                pointing_to : '../lib/systemd/systemd',
-                install_dir : sbindir)
-
 ############################################################
 
 core_test_template = test_template + {
diff --git a/src/hwdb/hwdb.c b/src/hwdb/hwdb.c
index 4d2d6fe8c4..4161259e49 100644
--- a/src/hwdb/hwdb.c
+++ b/src/hwdb/hwdb.c
@@ -35,7 +35,7 @@ static int help(void) {
         _cleanup_(table_unrefp) Table *options = NULL, *verbs = NULL;
         int r;
 
-        r = terminal_urlify_man("systemd-hwdb", "8", &link);
+        r = terminal_urlify_man("udev-hwdb", "8", &link);
         if (r < 0)
                 return log_oom();
 
diff --git a/src/hwdb/meson.build b/src/hwdb/meson.build
index 57ace3ad44..96c0b185d0 100644
--- a/src/hwdb/meson.build
+++ b/src/hwdb/meson.build
@@ -2,7 +2,7 @@
 
 executables += [
         executable_template + {
-                'name' : 'systemd-hwdb',
+                'name' : 'udev-hwdb',
                 'public' : true,
                 'conditions' : ['ENABLE_HWDB'],
                 'sources' : files('hwdb.c'),
diff --git a/src/libsystemd/sd-hwdb/sd-hwdb.c b/src/libsystemd/sd-hwdb/sd-hwdb.c
index 938d02f3b7..1640785361 100644
--- a/src/libsystemd/sd-hwdb/sd-hwdb.c
+++ b/src/libsystemd/sd-hwdb/sd-hwdb.c
@@ -413,7 +413,7 @@ static int hwdb_new(const char *path, sd_hwdb **ret) {
 
                 if (!hwdb->f)
                         return log_debug_errno(SYNTHETIC_ERRNO(ENOENT),
-                                               "hwdb.bin does not exist, please run 'systemd-hwdb update'.");
+                                               "hwdb.bin does not exist, please run 'udev-hwdb update'.");
         }
 
         if (fstat(fileno(hwdb->f), &hwdb->st) < 0)
diff --git a/src/test/meson.build b/src/test/meson.build
index 5d96b2fe74..978dfa5edf 100644
--- a/src/test/meson.build
+++ b/src/test/meson.build
@@ -26,18 +26,6 @@ endif
 
 generate_sym_test_py = files('generate-sym-test.py')
 
-test_libsystemd_sym_c = custom_target(
-        input : [libsystemd_sym_path,
-                 systemd_headers,
-                 libsystemd_sources],
-        output : 'test-libsystemd-sym.c',
-        command : [generate_sym_test_py,
-                   libsystemd_sym_path,
-                   libsystemd_dir_path,
-                   systemd_headers],
-        capture : true,
-        build_by_default : want_tests != 'false')
-
 test_libudev_sym_c = custom_target(
         input : [libudev_sym_path,
                  libudev_h_path,
@@ -50,7 +38,7 @@ test_libudev_sym_c = custom_target(
         capture : true,
         build_by_default : want_tests != 'false')
 
-generated_sources += [test_libsystemd_sym_c, test_libudev_sym_c]
+generated_sources += [test_libudev_sym_c]
 
 ############################################################
 
@@ -425,11 +413,6 @@ executables += [
                 'sources' : files('test-reread-partition-table-manual.c'),
                 'type' : 'manual',
         },
-        test_template + {
-                'sources' : files('test-sbat.c'),
-                'conditions' : ['ENABLE_BOOTLOADER'],
-                'c_args' : f'-I@efi_config_h_dir@',
-        },
         test_template + {
                 'sources' : files('test-seccomp.c'),
                 'dependencies' : libseccomp_cflags,
@@ -479,12 +462,6 @@ executables += [
         },
 
         # Symbol tests
-        test_template + {
-                'name' : 'test-libsystemd-sym',
-                'sources' : [test_libsystemd_sym_c],
-                'link_with' : libsystemd,
-                'suite' : 'libsystemd',
-        },
         test_template + {
                 'name' : 'test-libudev-sym',
                 'sources' : [test_libudev_sym_c],
diff --git a/src/udev/meson.build b/src/udev/meson.build
index b5140a626a..cb8fc3097a 100644
--- a/src/udev/meson.build
+++ b/src/udev/meson.build
@@ -256,7 +256,7 @@ executables += [
         },
 ]
 
-install_symlink('systemd-udevd',
+install_symlink('udevd',
                 pointing_to : libexecdir_to_bin + 'udevadm',
                 install_dir : libexecdir,
                 install_tag : 'udev')
diff --git a/src/udev/udev-config.c b/src/udev/udev-config.c
index c108985c4d..50b00f1b2f 100644
--- a/src/udev/udev-config.c
+++ b/src/udev/udev-config.c
@@ -165,8 +165,6 @@ static int help(void) {
         r = table_print_or_warn(options);
         if (r < 0)
                 return r;
-
-        help_man_page_reference("systemd-udevd.service", "8");
         return 0;
 }
 
diff --git a/src/udev/udevadm-hwdb.c b/src/udev/udevadm-hwdb.c
index 0a38e6ee06..0b4a09154d 100644
--- a/src/udev/udevadm-hwdb.c
+++ b/src/udev/udevadm-hwdb.c
@@ -32,7 +32,7 @@ static int help(void) {
 
         printf("\nNOTE:\n"
                "The sub-command 'hwdb' is deprecated, and is left for backwards compatibility.\n"
-               "Please use systemd-hwdb instead.\n");
+               "Please use udev-hwdb instead.\n");
         return 0;
 }
 
@@ -90,7 +90,7 @@ int verb_hwdb_main(int argc, char *argv[], uintptr_t _data, void *userdata) {
                 return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
                                        "Either --update or --test must be used.");
 
-        log_notice("udevadm hwdb is deprecated. Use systemd-hwdb instead.");
+        log_notice("udevadm hwdb is deprecated. Use udev-hwdb instead.");
 
         if (arg_update && !hwdb_bypass()) {
                 r = hwdb_update(arg_root, arg_hwdb_bin_dir, arg_strict, true);
diff --git a/src/udev/udevadm-wait.c b/src/udev/udevadm-wait.c
index 08c142d5eb..54d57f98d7 100644
--- a/src/udev/udevadm-wait.c
+++ b/src/udev/udevadm-wait.c
@@ -343,7 +343,7 @@ static int parse_argv(int argc, char *argv[]) {
                         break;
 
                 OPTION_LONG("initialized", "BOOL",
-                            "Wait for devices being initialized by systemd-udevd"):
+                            "Wait for devices being initialized by udevd"):
                         r = parse_boolean(opts.arg);
                         if (r < 0)
                                 return log_error_errno(r, "Failed to parse --initialized= parameter: %s", opts.arg);
diff --git a/src/udev/udevd.c b/src/udev/udevd.c
index 4cbc481c76..ca2e207ead 100644
--- a/src/udev/udevd.c
+++ b/src/udev/udevd.c
@@ -67,7 +67,7 @@ int run_udevd(int argc, char *argv[]) {
         if (arg_daemonize) {
                 pid_t pid;
 
-                log_info("Starting systemd-udevd version " GIT_VERSION);
+                log_info("Starting udevd version " GIT_VERSION);
 
                 /* connect /dev/null to stdin, stdout, stderr */
                 if (log_get_max_level() < LOG_DEBUG) {
diff --git a/test/meson.build b/test/meson.build
index fd3dba3a60..8284b24615 100644
--- a/test/meson.build
+++ b/test/meson.build
@@ -40,7 +40,7 @@ endif
 ############################################################
 
 if want_tests != 'false' and conf.get('ENABLE_HWDB') == 1
-        exe = executables_by_name.get('systemd-hwdb')
+        exe = executables_by_name.get('udev-hwdb')
         test('hwdb-test',
              files('hwdb-test.sh'),
              suite : 'dist',
-- 
2.55.0

